### ISSUE TITLE
`mix()` not generating hashes on `mix watch`

### ISSUE THREAD

**Issue by ojvribeiro** [dsktschy/laravel-mix-ejs#6]
First of all, thank you for this incredible extension ♥

-   Laravel Mix Version: `6.0.49`
-   Node Version: `v16.16.0`
-   NPM Version: `6.14.9`
-   OS: `Windows 10 Pro 21H2 19044.2006`

### Description:
I'm trying to setup a standalone project with Vue + Laravel Mix. I'm using [`laravel-mix-ejs`](https://laravel-mix.com/extensions/ejs) to add hashes to files with the `mix` function like so:

```html
<script src="<%= mix('/build/main.js') %>"></script>
``` 

When I run the `mix --production` command everything works just fine and the hashes are generated and appended to script tag, but when I run the `mix watch` command, new hashes are generated on the `mix-manifest.json` file, but the script path keeps just the plain path like so:

```html
<script src="/build/main.js"></script>
``` 

### Steps To Reproduce:
I made a repro on Stackblitz [here](https://stackblitz.com/edit/node-nuacts?file=index.ejs&view=editor)

- Run `npm run prod` on the terminal
- Check the script tag in the `build/index.html` file (works)
-
- Run `npm run watch`
- Check the script tag in the `build/index.html` file (no hashes)

**Comment by dsktschy** [comment]
@ojvribeiro 
Thank you for using and reporting ❤️

Hash output is currently not supported in development mode, so please try the following configuration instead.
```js
const mix = require('laravel-mix')
require('laravel-mix-ejs')

mix
  .setPublicPath('dist')
  .version()
  .js('src/js/app.js', 'dist/js')
  .sass('src/sass/app.scss', 'dist/css')
  .ejs(
    ['src/templates', 'dist/mix-manifest.json'],
    'dist',
    {},
    {
      base: 'src/templates',
      partials: ['dist/mix-manifest.json'],
      mixVersioning: true
    }
  )
```
I am considering supporting development mode as well, but until then, please use this way.

**Comment by ojvribeiro** [comment]
I appreciate your response.

Unfortunately I'm not able to get this running in my setup. Can you kindly explain what that configuration does?

Could you please fork [this Stackblitz minimal reproduction](https://stackblitz.com/edit/node-nuacts?file=index.ejs&view=editor) of my project? This is my basic structure:

```
📂build
┣ 📂css
┃ 📂js
┗ 📜index.html
📜index.ejs
📜mix-manifest.json
📜webpack.mix.js
```

**Comment by dsktschy** [comment]
@ojvribeiro
Could you try the following configration on your Windows machine instead of StackBlitz? The cause has not been confirmed, but on StackBlitz, hash output fails during watching. After confirming the expected behavior in your environment, I will explain the configration.
```js
const mix = require('laravel-mix')
require('laravel-mix-ejs')

mix
  .setPublicPath('.')
  .version()
  .js('main.js', 'build')
  .ejs(
    ['index.ejs', 'mix-manifest.json'],
    'build',
    {},
    {
      base: '.',
      partials: ['mix-manifest.json'],
      mixVersioning: true
    }
  )
```

**Comment by ojvribeiro** [comment]
It worked just fine on Windows 😄 

Thank you again!

**Comment by dsktschy** [comment]
@ojvribeiro
That's good!

Below is an explanation of the configuration.
```js
  .ejs(
    // Compile EJS even when `mix-manifest.json` is changed
    ['index.ejs', 'mix-manifest.json'],
    'build',
    {},
    {
      base: '.',
      // Do not output `mix-manifest.json` when EJS compilation
      partials: ['mix-manifest.json'],
      // The `mix` function outputs hash even in development mode
      mixVersioning: true
    }
  )
```

**Comment by ojvribeiro** [comment]
Oh I got it!

Again, thanks a lot for the workaround ❤️

**Comment by ojvribeiro** [comment]
Hi again, @dsktschy! I don't know if I should open a new issue for this, but suddenly (in `v2.0.1`) `prod` command stopped generanting the HTML file. I'm using the following code:

```js
mix
.ejs(
  [`${packagePath}/src/index.ejs`, `${rootPath}/_dist/mix-manifest.json`],
  `${rootPath}/_dist`,
  VulmixConfig,
  {
    partials: [`${rootPath}/_dist/mix-manifest.json`],
    mixVersioning: true,
  }
)
```

The EJS file comes from my package in `node_modules`, for context. The `mix-manifest.json` file with the hashes is generated, but not the `index.html` file:

![image](https://user-images.githubusercontent.com/8026741/234448376-9c0a26df-02f7-4c0e-a700-94504fb49164.png)

For some reason, the `watch --hot` command works just fine generating the index.html file.

**Comment by dsktschy** [comment]
@ojvribeiro 
Sorry, I could not reproduce it. Can you reproduce it on StackBlitz? In StackBlitz the `watch` command will fail, but the `build` command may succeed.
> The cause has not been confirmed, but on StackBlitz, hash output fails during watching.

**Comment by ojvribeiro** [comment]
Weird... On StackBlitz it just... works.

https://stackblitz.com/edit/node-7n8fef?file=app.vue

Run the `yarn prod` command. It will generate the `_dist` folder with the `index.html` file in it. On Windows tho, no success. I'll try to reinstall `node_modules`.

----

And yes, your lib is an important part of my meta-framework 😁

**Comment by ojvribeiro** [comment]
> I'll try to reinstall node_modules.

Nope, no success with that. :/

**Comment by dsktschy** [comment]
@ojvribeiro 
> https://stackblitz.com/edit/node-7n8fef?file=app.vue

`require('laravel-mix-ejs')` not found anywhere.

> And yes, your lib is an important part of my meta-framework 😁

Thank you for using ❤️

**Comment by ojvribeiro** [comment]
> `require('laravel-mix-ejs')` not found anywhere.

It's because it is in the `vulmix` package. 😁 Vulmix works like a Laravel Mix plugin as well.

----

Edit: Maybe you can clone the [`vulmix` repo](https://github.com/ojvribeiro/vulmix) and run the `prod` command for a better reproduction. The `index.html` file should be generated inside `demo/_dist`. As I said before, it works fine with the `dev` command (which uses Hot Module Replacement), but not with the `prod` command.

**Comment by dsktschy** [comment]
@ojvribeiro 
In the case of normal usage of laravel-mix-ejs (without vulmix), is the output as expected even in your environment? If so, the problem may lie in the difference between normal usage and the problem state. You can explore it inserting `console.log` in node_modules/laravel-mix-ejs. Pull requests are always welcome!

**Comment by ojvribeiro** [comment]
I found the problem, it was indeed an implementation bug. Apparently, I can't pass absolute paths to `laravel-mix-ejs`'s options, only relative paths. So I was previously trying to get the paths with `node:path`'s `path.resolve`, but it only returns absolute paths in Windows. So I created this function to convert absolute paths into relative paths:

```js
const path = require('node:path')

module.exports = function getRelativePath(from, to) {
  const relativePath = path.relative(from, to)

  if (relativePath !== '' && !from.match('node_modules\\vulmix')) {
    return './' + relativePath.replace(/\\/gi, '/')
  }

  return '.'
}
```

Now everything works fine.

Thanks a lot for pointing me in the right direction. ♥
